modules/search-service-impl/src/main/java/org/opencastproject/search/impl/persistence/SearchServiceDatabaseImpl.java

```diff
@@ -173,6 +173,27 @@ public void deleteMediaPackage(String mediaPackageId, Date deletionDate) throws
     }
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.search.impl.persistence.SearchServiceDatabase#countMediaPackages()
+   */
+  @Override
+  public int countMediaPackages() throws SearchServiceDatabaseException {
+    EntityManager em = emf.createEntityManager();
+    Query query = em.createNamedQuery("Search.getCount");
+    try {
+      Long total = (Long) query.getSingleResult();
+      return total.intValue();
```

Comment:
Why cast this to an `int`. This really shouldn't break anything given that it's unlikely that you have more than `MAX_INT` items in the database, but still, why not just stay with `long`?

Reply:
It's just copied from a similar implementation for series: https://github.com/opencast/opencast/blob/develop/modules/series-service-impl/src/main/java/org/opencastproject/series/impl/persistence/SeriesServiceDatabaseImpl.java#L535 

No objection to leaving it as long. Should I change it?

Reply:
Let's put it this way: Do you want to quickly update this? If not, then I'll hit merge :-)

Reply:
Merge :-) Hopefully these modules that use solr indexes will get refactored anyway to use elasticsearch indexes in due course, so this code won't be around for ever.